### PR TITLE
Add support for CSV output of `check` command. Closes #479

### LINKED ISSUE #479: csv output format

### PR DESCRIPTION
**Pull request by binaek89**

### CODE REVIEW

File: cmd/check.go [hunk 1]
```diff
@@ -128,10 +128,12 @@ func runCheckCmd(cmd *cobra.Command, args []string) {
 
 func validateOutputFormat() error {
 	outputFormat := viper.GetString(constants.ArgOutput)
-	if !helpers.StringSliceContains([]string{"text", "brief", "json", "none"}, outputFormat) {
-		return fmt.Errorf("invalid output format '%s' - must be one of json, text, brief, none", outputFormat)
+	// try to get a formatter for the desired output.
+	if _, err := controldisplay.GetFormatter(outputFormat); err != nil {
```
Comment (kaidaguerre) [inline cmd/check.go:132]: what error is shown for an invalid output arg? - screenshot?

File: control/controldisplay/csv.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+import (
+	"log"
+
+	"github.com/turbot/go-kit/helpers"
+	typehelpers "github.com/turbot/go-kit/types"
+	"github.com/turbot/steampipe/control/execute"
+)
+
+type GroupCsvRenderer struct {
```
Comment (kaidaguerre) [inline control/controldisplay/csv.go:11]: I think we can just call it `CSVControlRenderer` - group is a bit generic (I know it renders resultgroups, but I think we can be more general in the name)

Comment (kaidaguerre) [reply control/controldisplay/csv.go:11]: capitalisation should match CSVFormatter

File: control/controldisplay/csv.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	return &GroupCsvRenderer{}
+}
+
+func (r GroupCsvRenderer) Render(tree *execute.ExecutionTree) [][]string {
+	r.columns = newResultColumns(tree)
+	return r.renderGroup(tree.Root)
+}
+
+func (r GroupCsvRenderer) renderGroup(group *execute.ResultGroup) [][]string {
+	log.Printf("[TRACE] begin group  csv render '%s'\n", group.GroupId)
```
Comment (kaidaguerre) [inline control/controldisplay/csv.go:25]: is this logging still needed?

File: control/controldisplay/formatter_text.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+	"github.com/karrick/gows"
+	"github.com/turbot/steampipe/control/execute"
+)
+
+type TextFormatter struct{}
+
+func (j *TextFormatter) Format(ctx context.Context, tree *execute.ExecutionTree) (io.Reader, error) {
+	// limit to 200
+	maxCols := j.getMaxCols(200)
```
Comment (kaidaguerre) [inline control/controldisplay/formatter_text.go:16]: use a const for the hard limit

**Review by kaidaguerre — CHANGES_REQUESTED** [review]

**Review by kaidaguerre — CHANGES_REQUESTED** [review]
Please add BATS test

File: cmd/check.go (hunk 1 above)
Comment (binaek89) [reply cmd/check.go:132]: Screenshot 2021-06-07 at 7 55 53 PM

**Review by kaidaguerre — CHANGES_REQUESTED** [review]

File: control/execute/control_run.go [hunk 5]
```diff
@@ -111,71 +110,78 @@ func (r *ControlRun) Start(ctx context.Context, client *db.Client) {
...
 			viper.Set(constants.ArgSearchPathPrefix, originalConfiguredSearchPathPrefix)
 			client.SetClientSearchPath()
 		}
+		close(gatherDoneChan)
 	}()
 
-	// TEMPORARY - we will eventually pass the streams to the renderer before completion
-	// wait for control to finish
-	controlCompletionTimeout := 240 * time.Second
 	for {
```
Comment (kaidaguerre) [inline control/execute/control_run.go:141]: can this just be 
```
	gatherDone := false
	for !gatherDone {		
		select {
		case <-ctx.Done():
			r.SetError(fmt.Errorf("control timed out"))
		case <-gatherDoneChan:
			gatherComplete = true
		}
	}
```

File: control/execute/control_run.go [hunk 6]
```diff
@@ -111,71 +110,78 @@ func (r *ControlRun) Start(ctx context.Context, client *db.Client) {
...
-				//log.Println("[WARN]", "set error", r.Control.ShortName)
 				r.SetError(err)
 				return
 			}
 			r.addResultRow(result)
 		case <-r.doneChan:
 			return
-		default:
-			time.Sleep(25 * time.Millisecond)
+			// default:
```
Comment (kaidaguerre) [inline control/execute/control_run.go:183]: delete

**Review by kaidaguerre — CHANGES_REQUESTED** [review]

File: control/execute/control_run.go [hunk 7]
```diff
@@ -111,71 +110,74 @@ func (r *ControlRun) Start(ctx context.Context, client *db.Client) {
...
-			r.SetError(fmt.Errorf("control %s timed out", control.Name()))
+	resultsGathered := false
+	for !resultsGathered {
+		select {
+		case <-ctx.Done():
+			r.SetError(fmt.Errorf("control timed out"))
+			resultsGathered = true
+		case <-gatherDoneChan:
+			resultsGathered = true
+		default:
```
Comment (kaidaguerre) [inline control/execute/control_run.go:149]: why the sleep?

**Pull request merged**